src/UI/researchScreen/ResearchScreen.ts

```diff
@@ -18,8 +19,15 @@ export default class ResearchScreen implements Page {
     refresh(): void {
         let researchHeader = UI.makeHeader("Available Research Projects");
 
-
         const possibleTechs: Technology[] = this.run.getResearchOptions();
+        const researchResources: Resource[] = Resource.values().filter(resource => possibleTechs.filter(tech => tech.researchCost.resource == resource).length != 0);
+
+        const researchResourcesHTML = UI.makeDivContaining(researchResources.length != 0
+            ? [
+                UI.makeHeader("Available Research Resources"),
+                ...researchResources.map(resource => UI.makePara(`${resource.name}: ${this.run.inventory.getResourceQuantity(resource)}`))
+            ]
```

Comment:
This puts a lot of different logic into one "line", so even though you wrap it across multiple lines it's kind of hard to read.

Maybe initialize reasearchResourcesHTML as an empty div, then only have it `appendchild` the header and the the resources if the array of research resources is nonempty. That way you wouldn't have to do complicated ternary logic inside your UI call.